### ISSUE TITLE
Only throw real exceptions

### ISSUE THREAD

**Issue by mglisse** [GUDHI/gudhi-devel#793]
In several places in gudhi, we have
```c++
    throw "some explanation";
```
However, if the exception is not caught, this only prints
> terminate called after throwing an instance of 'char const*'

So it does not convey more information than say
```c++
    throw 42; // some explanation
```

It would be more useful to encapsulate this text in some real exception object (from `<stdexcept>` for instance). I think it would also interact better with Python exceptions through cython or pybind11.

**Comment by VincentRouvreau** [comment]
I completely agree with you. I like the way it is advised in [this c++ tutorial](https://openclassrooms.com/fr/courses/7137751-programmez-en-oriente-objet-avec-c/7532931-gerez-des-erreurs-avec-les-exceptions#/id/r-7533183) (in french).
Do you think something equivalent could be added to the [C++ code conventions](https://github.com/GUDHI/gudhi-devel/blob/master/.github/code_conventions.md) ?

**Comment by VincentRouvreau** [comment]
We should also keep in mind the way [cython translates C++ exceptions in python](https://cython.readthedocs.io/en/stable/src/userguide/wrapping_CPlusPlus.html?highlight=io#exceptions)

**Comment by VincentRouvreau** [comment]
*[UPDATE Sept 2026]* I can see:
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Bitmap_cubical_complex/include/gudhi/Bitmap_cubical_complex.h#L272-L273
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Coxeter_triangulation/include/gudhi/Functions/Constant_function.h#L42
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Persistence_representations/include/gudhi/Persistence_intervals.h#L241
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Persistence_representations/include/gudhi/Persistence_landscape_on_grid.h#L550
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Persistence_representations/include/gudhi/Persistence_landscape_on_grid.h#L694
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Persistence_representations/include/gudhi/Persistence_landscape_on_grid.h#L987
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Persistence_representations/include/gudhi/Persistence_vectors.h#L110
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Persistence_representations/include/gudhi/Persistence_vectors.h#L354
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Rips_complex/example/example_one_skeleton_rips_from_correlation_matrix.cpp#L40
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/Rips_complex/utilities/rips_correlation_matrix_persistence.cpp#L59
* https://github.com/GUDHI/gudhi-devel/blob/1ee115515e256ab37700a539c277713755590834/src/common/include/gudhi/Simple_object_pool.h#L56

Maybe not all exceptions need to be fixed. The ones generated by an example, for instance.
